File: dev/FileScanner/group.py
from datetime import datetime

class InvalidTimestampFormatException(Exception):
    pass

class Group:
    def __init__(self, user_email, timestamp, number_downloads, group_uuid=None):
        self.group_uuid = group_uuid
        self.user_email = user_email
        self.timestamp = self.validate_and_format_timestamp(timestamp)
        self.number_downloads = number_downloads

    def get_group_uuid(self):
        return self.group_uuid

    def get_user_email(self):
        return self.user_email

    def get_timestamp(self):
        return self.timestamp

    def get_number_downloads(self):
        return self.number_downloads

    @staticmethod
    def validate_and_format_timestamp(timestamp):
        try:
            # Parse the input timestamp to validate and format it
            datetime_object = datetime.fromisoformat(timestamp)

            # Return the formatted timestamp as a string
            return datetime_object.isoformat()
        except ValueError:
            raise InvalidTimestampFormatException("Invalid timestamp format. Expected format: YYYY-MM-DDTHH:MM:SS")

File: app/src/main/java/dev/test/trustisend/entity/Group.java
package dev.test.trustisend.entity;

import dev.test.trustisend.exception.InvalidTimestampFormatException;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.time.format.DateTimeParseException;

public class Group {
    private String groupUUID;
    private String userEmail;
    private String timestamp;
    private Integer numberDownloads;

    public Group(String userEmail,  String timestamp, Integer numberDownloads) {
        this.userEmail = userEmail;
        this.timestamp = validateAndFormatTimestamp(timestamp);
        this.numberDownloads = numberDownloads;
    }

    public Group(String groupUUID, String userEmail, String timestamp, Integer numberDownloads) {
        this.groupUUID = groupUUID;
        this.userEmail = userEmail;
        this.timestamp = validateAndFormatTimestamp(timestamp);
        this.numberDownloads = numberDownloads;
    }

    public String getGroupUUID() {
        return groupUUID;
    }

    public String getUserEmail() {
        return userEmail;
    }

    public String getTimestamp() {
        return timestamp;
    }

    public Integer getNumberDownloads() {
        return numberDownloads;
    }


    private String validateAndFormatTimestamp(String timestamp) {
        try {
            // Define the expected timestamp format
            DateTimeFormatter formatter = DateTimeFormatter.ISO_DATE_TIME;

            // Parse the input timestamp to validate and format it
            LocalDateTime dateTime = LocalDateTime.parse(timestamp, formatter);

            // Return the formatted timestamp as a string
            return dateTime.format(formatter);
        } catch (DateTimeParseException e) {
            throw new InvalidTimestampFormatException("Invalid timestamp format. Expected format: YYYY-MM-DDTHH:MM:SS");
        }
    }
}
